pkg/policy/api/egress.go

```diff
@@ -176,6 +180,29 @@ func (e *EgressRule) GetDestinationEndpointSelectors() EndpointSelectorSlice {
 	return e.aggregatedSelectors
 }
 
+// GetDestinationEndpointSelectorsWithRequirements returns a slice of endpoints selectors covering
+// all L3 source selectors of the ingress rule
```

Comment:
I suspect that with the rest of the other PR eliminating dead code, we might be able to fold `GetDestinationEndpointSelectorsWithRequirements()` and `GetDestinationEndpointSelectors()` together?

Also:
```suggestion
// all L3 destination selectors of the egress rule
```

Reply:
Looked into this, and indeed, `canReach*()` functions as well as `GetDestinationEndpointSelectors()` can be removed. I'll prepare a separate commit for this in another PR.

Reply:
Folded these dead code removals here to avoid unnecessary work fixing the dead functions.